docker/api/build.py

```diff
@@ -46,7 +46,9 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if os.path.exists(dockerignore):
                 with open(dockerignore, 'r') as f:
                     exclude = list(filter(bool, f.read().splitlines()))
-            context = utils.tar(path, exclude=exclude, dockerfile=dockerfile)
+            context = utils.tar(path, exclude=exclude, dockerfile=dockerfile,
+                gzip=True)
+            encoding = 'gizp'
```

Comment:
I think there should be a kwarg `gzip_context` (default to `False`) to enable it. There may be cases where it's undesirables, and defaulting it to false keeps it backwards compatible.

The new kwarg will need a small docs addition.